tensorflow_datasets/scripts/cli/list.py

```diff
@@ -0,0 +1,23 @@
+
+"""`tfds list` command."""
+
+import argparse
+
+from tensorflow_datasets.core import load
+
+
+def register_subparser(parsers: argparse._SubParsersAction) -> None:  # pylint: disable=protected-access
+  """Add subparser for `list` command."""
+  new_parser = parsers.add_parser('list', help='Lists all datasets')
+  new_parser.set_defaults(subparser_fn=_list_datasets)
+
+
+def _list_datasets(args: argparse.Namespace) -> None:
+  """Creates the dataset directory. Executed by `tfds new <name>`."""
```

Comment:
This should be modified for `tfds list` command

Reply:
Not sure what you mean but this will be heavily changed for the new flags. This function's purpose is to handle the `args` (including validation) and change control flow from here on. No other function should have to deal with `args`. Please review it again then!